docs/reference/mapping/params/subobjects.asciidoc

```diff
@@ -109,26 +109,136 @@ PUT my-index-000001/_doc/metric_1
 <1> The entire mapping is configured to not support objects.
 <2> The document does not support objects
 
+Setting `subobjects: false` disallows the definition of <<object,`object`>> and <<object,`nested`>> sub-fields, which
+can be too restrictive in cases where it's desirable to have nested objects or sub-objects with specific behavior
+(e.g. with `enabled:false`). In this case, it's possible to set `subobjects: auto`, so subobjects are allowed to be
+statically defined in the mapping, while dynamically-defined fields do not introduce intermediate objects. For instance:
```

Comment:
A nuance that's missing here is that statically defined mappers are also auto-flattened if possible.

Reply:
I think this is covered in the section below?

Reply:
It is, I was just thinking it may be confusing as it seems to not be consistent with what's stated below. Both statically and dynamically defined fields are auto-flattened unless conditions apply where auto-flattening isn't possible. I think these condition may even apply to dynamically added fields, for example if a dynamic template defines an object with `enabled:false`. (it may be worth adding a test for this)


With a dynamic template like this
```json
{
  "mappings": {
    "subobjects": "auto",
    "dynamic_templates": [
      {
        "foo_object": {
          "match": "foo",
          "mapping": {
            "type": "object",
            "enabled": false
          }
        }
      }
    ]
  }
}
``` 

Adding a field `foo.bar` should create a `foo` object, whereas adding a field `bar.baz` should not.

Reply:
Good point, added an explicit reference to dynamic templates and a yaml test to verify this example works.